submissions/evgeniy24/friends_app/app.js

```diff
@@ -0,0 +1,234 @@
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_NAME_ZA) {
+        sortedList.sort((a, b) => sortByName(b, a));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_AGE_YO) {
+        sortedList.sort((a, b) => sortByAge(a, b));
+        renderFriendsList(sortedList);
+
+    } else if (event.target === SORT_BY_AGE_OY) {
+        sortedList.sort((a, b) => sortByAge(b, a));
+        renderFriendsList(sortedList);
+    } 
+})
+
+//filtering list of friends by gender
+FILTER_MENU.addEventListener('click', function(event) {
+    if (searchResultList) {
+        genderNeutralList = searchResultList;
+    } else {
+        genderNeutralList = [...friendsList];
+    }
+
+    if (event.target === FILTER_BY_MALE) {
+        filteredByGenderList = genderNeutralList.filter(function(friend) {
+           return friend.gender === 'male'; 
+        })
+        renderFriendsList(filteredByGenderList);
+        
+    } else if (event.target === FILTER_BY_FEMALE) {
+        filteredByGenderList = genderNeutralList.filter(function(friend) {
+           return friend.gender === 'female'; 
+        })
+        renderFriendsList(filteredByGenderList);
+
+    } else if (event.target === FILTER_BY_ALL) {
+        filteredByGenderList = [...friendsList];
+        renderFriendsList(filteredByGenderList);
+        }
+        
+})
+
+// showing user message when data loading error ------------------------------------------------------
+function renderErrorMessage() {
+    const messageErrorWrap =  document.createElement('div');
+    const messageError = document.createElement('p');
+    const tryMoreBtn = document.createElement('button');
+
+    messageErrorWrap.classList.add('error-message-wrap')
+    tryMoreBtn.classList.add('btn');
+    tryMoreBtn.classList.add('error-btn');
+    messageError.classList.add('error-msg-text');
+
+    messageError.textContent = 'OOPS! Something Bad Happened, please try again';
+    tryMoreBtn.textContent = 'Try Again';
+
+
+    messageErrorWrap.append(messageError, tryMoreBtn);
+    FRIENDS_LIST_WRAP.append(messageErrorWrap);
+}
+
+// show/hide filter and sorting menus on mobile devices
+MOBILE_MENU_BUTTON.addEventListener('click', function() {
+    clickMenuBtn(MOBILE_MENU_BUTTON);
+    clickMenuBtn(MENU_WRAP);
+    scrollDepth = window.pageYOffset;
+    scrollToTop();
+});
+
+function clickMenuBtn(elem) {
+    elem.classList.toggle('change');
+}
+
+//smooth scrolling to top
+function scrollToTop() {   
+    let timer = 0; 
+    if (scrollDepth > 100) {
+        window.scrollTo(pageXOffset, scrollDepth);
+        scrollDepth = scrollDepth - 60;
+        timer = setTimeout(scrollToTop, 15);
+    } else {
+        clearTimeout(timer);
+        window.scrollTo(0,0);
+    }
+   
+} 
+
+// when error - reSend Request to server -----------------------
+FRIENDS_LIST_WRAP.addEventListener('click', function(event) {
+    if (event.target.classList == 'btn error-btn') {
```

Comment:
What if classname for error will be changed? 🤔

Reply:
Done, refactored ! See a new commit!